ui/js/controllers/bugfiler.js

```diff
@@ -127,8 +127,13 @@ treeherder.controller('BugFilerCtrl', [
             summary = summary.split(" | ");
 
             for (var i=0; i < $scope.omittedLeads.length; i++) {
-                if (summary[0].search($scope.omittedLeads[i]) >= 0 && summary.length > 1) {
-                    summary.shift();
+                // In the case where only exact substring matches will provide usable bugs,
+                // Dropping things from omittedLeads will cause the substring to not match.
+                // Don't drop the lead in this case.
+                if ($scope.search_terms.length > 0 && !$scope.search_terms[0].includes($scope.omittedLeads[i])) {
```

Comment:
A job can have multiple search_terms defined, but the bug filer is only sent the search_terms for a single failure line. 

That single search_terms is an array with zero or more items contained within it, but the only time one of the omitted leads should be present is when a good search_term could be found, so we just use the failure line as a whole (capped at 100 characters, I believe). 

The only time multiple items should be within search_terms is if we actually found valid things (multiple filenames, crash signatures, etc).

I can add the comment and combine the if()s, for sure.